docs/painless/painless-guide/painless-execute-script.asciidoc

```diff
@@ -154,13 +249,560 @@ POST /_scripts/painless/_execute
+`2000-03-15`, the script calculates that the author started writing
+`Revelation Space` on 19 February 1999. Writing a 585 page book in just over one
+year is pretty impressive!
+
+[source,console-result]
+----
+{
+  "result" : [
+    "1999-02-19T00:00:00.000Z"
+  ]
+}
+----
+
+[[painless-runtime-double]]
+===== `double_field`
+Use the `double_field` runtime context for {ref}/number.html[numeric data] of type
+`double`. For example, let's say you have sensor data that includes a `voltage`
+field with values like 5.6. You need to multiply this value, but only for
+sensors that match a specific model number.
+
+Add the following fields to your index mapping. The `voltage` field is a 
+sub-field of the `measures` object.
+
+[source,console]
+----
+PUT my-index-000001
+{
+  "mappings": {
+    "properties": {
+      "@timestamp": {
+        "type": "date"
+      },
+      "model_number": {
+        "type": "keyword"
+      },
+      "measures": {
+        "properties": {
+          "voltage": {
+            "type": "double"
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+The following script matches on any documents where the `model_number` equals
+`QVKC92Q`, and then multiplies the `voltage` value by `1.7`. This script is
+useful when you want to select specific documents and only operate on values
+that match the specified criteria.
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """if (doc['model_number'].value.equals('QVKC92Q'))
+        {emit(1.7 * params._source['measures']['voltage']);}
+        else{emit(params._source['measures']['voltage']);}"""
+  },
+  "context": "double_field",
+  "context_setup": {
+    "index": "my-index-000001",
+    "document": {
+      "@timestamp": 1516470094000,
+      "model_number": "QVKC92Q",
+      "measures": {
+        "voltage": 5.6
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+The result includes the calculated voltage, which was determined by multiplying
+the original value of `5.6` by `1.7`:
+
+[source,console-result]
+----
+{
+  "result" : [
+    9.52
+  ]
+}
+----
+
+[[painless-runtime-geo]]
+===== `geo_point_field`
+{ref}/geo-point.html[Geo point] fields accept latitude-longitude pairs. You can define a
+geo-point field in several ways, and include values for latitude and longitude
+in the document for your script.
+
+If you already have a known geo-point, it's simpler to clearly state the
+positions of `lat` and `lon` in your index mappings. 
+
+[source,console]
+----
+PUT /my-index-000001/
+{
+  "mappings": {
+    "properties": {
+      "lat": {
+        "type": "double"
+      },
+     "lon": {
+        "type": "double"
+      }
+    }
+  }
+}
+----
+
+You can then use the `geo_point_field` runtime field context to write a script
+that retrieves the `lat` and `long` values.
+
+[source,console]
+----
+POST /_scripts/painless/_execute
+{
+  "script": {
+    "source": """
+        emit(doc['lat'].value, doc['lon'].value);
```

Comment:
This still uses the old style "noop" script. We'd talked about waiting on this one for later. Is that still what you are thinking?

Reply:
I talked to Ignacio, and he said that this is the preferred method. I originally included a `location` object in the mappings, and then used `emit(doc['location'].lat, doc['location'].lon);` in the script. Ignacio said that it's better if the original mapping includes `lat` and `lon` fields mapped as `double` so that you can use them to emit a geopoint. This was my original:

```
PUT /my-index-000001/
{
  "mappings": {
    "properties": {
      "location": {
        "type": "geo_point"
      }
    }
  }
}
POST /_scripts/painless/_execute
{
  "script": {
    "source": """
        emit(doc['location'].lat, doc['location'].lon);
        """
  },
  "context": "geo_point_field",
  "context_setup": {
    "index": "my-index-000001",
    "document": {
      "location": {
        "lat": 41.12,
        "lon": -71.34
      }
    }
  }
}
```